mongodbatlas/data_source_mongodbatlas_data_lake.go

```diff
@@ -0,0 +1,136 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func dataSourceMongoDBAtlasDataLake() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceMongoDBAtlasDataLakeRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"aws_role_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"aws_test_s3_bucket": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"aws_iam_assumed_role_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"aws_iam_user_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"aws_external_id": {
+				Type:     schema.TypeString,
+				Computed: true,
```

Comment:
I did that to be more readable and since it can be separate with provider name like this `aws_` but I can try to put under `cloudProviderConfig` if you want @andreaangiolillo

Reply:
oh no no, I agree with you that this is easier to read but I was wondering if we should have some guidelines that explicitly say when to apply this process.

Reply:
I see, yes it would be nice to have some guidelines similar to CONTRIBUTING.md of aws provider

Reply:
yea, not enough time. okay Edgar let's move this into an aws block. remember when we don't plan in case we add another provider azure, gcp, we end up with a mess when doing diffs. and the logic is ugly.

Reply:
@andreaangiolillo , @coderGo93 , the plan is to use the wiki https://github.com/mongodb/terraform-provider-mongodbatlas/wiki (which is empty) to specify this kind of things.

Reply:
Got it! Thanks!

Reply:
changed to aws block, @andreaangiolillo @leofigy